tensorrt_llm/_torch/auto_deploy/shim/ad_executor.py

```diff
@@ -307,6 +601,108 @@ def forward(
         return {"logits": logits_flat}
 
 
+def create_draft_model_engine_maybe(
+    ad_config: LlmArgs,
+    engine,
+    dist_mapping: Mapping,
+    mpi_dist: MPIDist,
+    max_draft_len: int,
+    max_total_draft_tokens: int,
+) -> Optional[PyTorchModelEngine]:
+    """Create a draft model engine for speculative decoding.
+
+    Args:
+        ad_config: The AutoDeploy LLM configuration
+        engine: The target model engine (ADEngine)
+        dist_mapping: The distributed mapping configuration
+        mpi_dist: The MPI distribution object
+        max_draft_len: Maximum draft length for speculative decoding
+        max_total_draft_tokens: Maximum total draft tokens
+
+    Returns:
+        PyTorchModelEngine configured as a draft model, or None if not needed
+    """
+    spec_config = ad_config.speculative_config
+    if spec_config is None or not spec_config.spec_dec_mode.has_draft_model():
+        return None
+
+    has_spec_drafter = spec_config.spec_dec_mode.has_spec_drafter()
+
+    draft_spec_config = copy.copy(spec_config)
+
+    use_chain_drafter = (
+        ad_config.guided_decoding_backend is None
+        and draft_spec_config._allow_chain_drafter
+        and draft_spec_config._allow_greedy_draft_tokens
+        and ad_config.attn_backend == "TRTLLM"
```

Comment:
this backend is also not used in `ad_config`. There is `flashinfer` and `triton`